File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;
class Solution {
    public int solution(int[] order) {
        Stack<Integer> aux = new Stack<>();
        int idx = 1;
        int cnt = 0;
        while (idx < order.length+1) {
            aux.push(idx); 
            while (!aux.isEmpty() && aux.peek() == order[cnt]) {
                cnt++;
                aux.pop();
            }
            idx++;
        }

        return cnt;
    }
}

File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    aux = []
    idx = 1
    cnt = 0
    while idx < len(order) + 1:
        aux.append(idx)
        while aux and aux[-1] == order[cnt]:
            cnt += 1
            aux.pop()
        idx += 1

    return cnt